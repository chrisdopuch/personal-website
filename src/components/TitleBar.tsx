import AppBar from '@material-ui/core/AppBar';
import IconButton from '@material-ui/core/IconButton';
import { createStyles, Theme, WithStyles, withStyles } from '@material-ui/core/styles';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import MenuIcon from '@material-ui/icons/Menu';
import React, { SFC } from 'react';
import { useStore } from 'react-hookstore';
import { IAppStore } from '../index';
import { appTheme } from './App';

const stylesDeclarations = (theme: Theme) => {
  return createStyles({
    appBar: {
      zIndex: theme.zIndex.drawer + 1,
    },
    menuButton: {
      marginRight: 20,
      [theme.breakpoints.up('md')]: {
        display: 'none',
      },
    },
  });
};

interface ITitleBarProps extends WithStyles<typeof stylesDeclarations, true> {
  title: string;
}

export const TitleBar: SFC<ITitleBarProps> = (props) => {
  const { classes, title } = props;
  const [appStore, setStore]: [IAppStore, (s: IAppStore) => void] = useStore('appStore');

  return (
    <AppBar position="fixed" className={classes.appBar}>
      <Toolbar>
        <IconButton
          color="inherit"
          aria-label="Open drawer"
          onClick={() =>
            setStore(
              Object.assign({}, appStore, {
                isNavDrawerOpen: !appStore.isNavDrawerOpen,
              })
            )
          }
          className={classes.menuButton}
        >
          <MenuIcon />
        </IconButton>
        <Typography variant="h6" color="inherit" noWrap={true}>
          {title}
        </Typography>
      </Toolbar>
    </AppBar>
  );
};

TitleBar.defaultProps = {
  theme: appTheme,
};

export default withStyles(stylesDeclarations, { withTheme: true })(TitleBar);
