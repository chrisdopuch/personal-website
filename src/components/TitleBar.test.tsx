import { IconButton, Theme } from '@material-ui/core';
import { shallow } from 'enzyme';
import React from 'react';
import { TitleBar } from './TitleBar';

const mockDispatch = jest.fn();
jest.mock('../hooks/useDispatch', () => {
  return jest.fn(() => mockDispatch);
});

function getDefaultProps() {
  return {
    classes: {
      appBar: 'appBar',
      menuButton: 'menuButton',
    },
    theme: {} as Theme,
    title: 'default title',
  };
}

describe('TitleBar', () => {
  it('renders default props', () => {
    const wrapper = shallow(<TitleBar {...getDefaultProps()} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("dispatches `toggleNavDrawerOpen` action on IconButton's `onClick`", () => {
    const wrapper = shallow(<TitleBar {...getDefaultProps()} />);

    wrapper
      .find(IconButton)
      .props()
      .onClick();

    expect(mockDispatch).toBeCalledWith({ type: 'toggleNavDrawerOpen' });
  });
});
