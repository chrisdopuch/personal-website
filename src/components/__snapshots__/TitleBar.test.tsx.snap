// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar snapshots renders default props 1`] = `
<WithStyles(AppBar)
  className="appBar"
  position="fixed"
>
  <WithStyles(Toolbar)>
    <WithStyles(IconButton)
      aria-label="Open drawer"
      className="menuButton"
      color="inherit"
      onClick={[Function]}
    >
      <pure(MenuIcon) />
    </WithStyles(IconButton)>
    <WithStyles(Typography)
      color="inherit"
      noWrap={true}
      variant="h6"
    >
      default title
    </WithStyles(Typography)>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
